lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -334,7 +334,7 @@ private Task InstallDMGAsync(string dmgPath, string folderPath)
                 process.BeginOutputReadLine();
                 process.WaitForExit();
 
-                return mountAndCopyTcs.Task.WithTimeout(Puppeteer.DefaultTimeout);
+                await mountAndCopyTcs.Task.WithTimeout(Puppeteer.DefaultTimeout).ConfigureAwait(false);
```

Comment:
To ensure the `finally` is executed _after_ `mountAndCopyTcs` is done or if it throws

Reply:
Good catch!